provisioner/clusterpy.go

```diff
@@ -849,8 +849,9 @@ func (p *clusterpyProvisioner) prepareProvision(logger *log.Entry, cluster *api.
 	if err != nil {
 		return nil, nil, err
 	}
+	crdResolver := NewKarpenterCRDResolver(cluster.ConfigItems["karpenter_main_crd_name"])
```

Comment:
What if we simply try the newest CRD first and fallback to the other one if it returns not found? Then we don't need to set a config-item and clean that up later and so on?

Reply:
after a second thought, I think it is better to inspect the cluster and automatically pick the right CRD